typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,51 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+// ProviderConfigProperties allows to get the additional properties. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
+// Reference: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest/test/additional-properties.test.ts
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("This configuration holds the necessary information to authenticate and interact with a provider for the recipe execution.")
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+  secrets?: Record<RecipeSecret>;
+}
+
+// EnvironmentVariables allows to get the additional properties. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
+// Reference: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest/test/additional-properties.test.ts
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("The environment variables injected during Terraform Recipe execution for the recipes in the environment.")
+model EnvironmentVariables extends Record<unknown> {
+  @doc("The provider secrets passed as environment variables.")
+  secrets?: Record<RecipeSecret>;
+}
+
+@doc("This secret is used within a recipe. Secrets are encrypted, often have fine-grained access control, auditing and are recommended to be used to hold sensitive data.")
+model RecipeSecret {
+  @doc("The ID of an Applications.Core/SecretStore resource containing sensitive data required for recipe execution.")
+  source: string;
+
+  @doc("The key for the secret in the secret store.")
```

Comment:
@vishwahiremat could you confirm if the property name for secretstore resource we use is `name` or `key`?

```suggestion
  @doc("The name for the secret in the secret store.")
```

Reply:
I think `key` is the right word here, because i think we are specifying the "name of the key" here that is stored in the secretstore.